### PR TITLE
SFB-213: Ttl code that has shacl errors is not passed on

### PR DESCRIPTION
**Pull request by JonasVanHoof**
## ID
 [SFB-213](https://binnenland.atlassian.net/browse/SFB-213)

 ## Description

When updatin gthe ttl code is often brekas the application so you cannot do anything anymore. To fix this im checking if the shacl shapes are returning any errors.

Also update the console messages to show errors and warnings differently.

 ## Type of change

 - [x] Bug fix
 - [x] New feature
 - [ ] Breaking change
 - [ ] Other

 ## How to test

1. Open the console it should display a difference between a warning and an error
2. Change the display type of a field. It will not update the preview as it is not valid
3. The check if there are shacl errors is only done for the code editor as this is the place where users can freely edit the ttl code

 ## Links to other PR's

 - /

### CODE REVIEW

File: app/templates/formbuilder/edit/code.hbs [hunk 1]
```diff
@@ -1,41 +1,39 @@
...
+              class={{message.severity.iconClass}}
+              @icon={{message.severity.icon}}
+              @size="large"
+              @alignment="left"
+            />
+            <p>{{message.subject}}</p>
+            <p class="au-u-bold">{{message.content}}</p>
+          </AuContent>
+        {{/each}}
+      </AuBodyContainer>
```
Comment (JonasVanHoof) [inline app/templates/formbuilder/edit/code.hbs:36]: Will look like this now 
image

we should definitly think about what we want to show in here and how we want to show it  as we cannot change the format everytime we have a new message to display.

File: app/controllers/formbuilder/edit/code.js [hunk 2]
```diff
@@ -34,17 +40,28 @@ export default class FormbuilderEditCodeController extends Controller {
...
+    if (this.ttlHasErrors()) {
+      this.consoleMessages.pushObject({
+        severity: this.getConsoleSeverity(SHACL_SEVERITY_TYPE.info),
+        subject: this.intl.t('messages.subjects.preview'),
+        content: this.intl.t(
+          'messages.feedback.previewNotUpdatedBecauseOfErrorsInTtl'
+        ),
+      });
+      return;
+    }
```
Comment (JonasVanHoof) [inline app/controllers/formbuilder/edit/code.js:59]: When there are errors the ttl code is not saved and put in the preview. This makes it that the application is protected for the shapes we have now. Ofcourse there are other ways to write invalid ttl code but this is a start where we can look at more and better shacl shapes.

Possible todo:
 * new/better shapes
 * checking if the predicates are valid
 * checking if the given object values are referenced in the ttl or of the expected type

This is best implemented in another PR so this one has one issue solved

File: app/controllers/formbuilder/edit/code.js [hunk 3]
```diff
@@ -55,18 +72,16 @@ export default class FormbuilderEditCodeController extends Controller {
...
     const formattedReport = formatShaclValidationReport(shaclReport);
+    console.warn('Formatted SHACL report: ', formattedReport);
+
     formattedReport.errorDetails.forEach((error) => {
-      this.warnings.push({
+      this.consoleMessages.pushObject({
+        severity: this.getConsoleSeverity(error.severity),
         subject: error.subject,
-        message: error.messages,
+        content: error.messages,
```
Comment (JonasVanHoof) [inline app/controllers/formbuilder/edit/code.js:84]: As we want to display syntax errors and maybe more this format needs to be as generic as possible. Create a utils object where you can pass data and it returns it as a console message

File: app/controllers/formbuilder/edit/code.js [hunk 4]
```diff
@@ -83,4 +98,49 @@ export default class FormbuilderEditCodeController extends Controller {
...
+        icon: 'alert-triangle',
+        class: baseClass + 'code-edit-message--bg-warning',
+        iconClass: 'code-edit-message--icon-warning',
+      },
+      [SHACL_SEVERITY_TYPE.info]: {
+        type: SHACL_SEVERITY_TYPE.info,
+        icon: 'info-circle',
+        class: baseClass + 'code-edit-message--bg-info',
+        iconClass: 'code-edit-message--icon-info',
+      },
```
Comment (JonasVanHoof) [inline app/controllers/formbuilder/edit/code.js:122]: Works with the values got from the SHACL report. Type stylings and icons are done like this as it was the only way I could think of doing this

File: app/controllers/formbuilder/edit/code.js [hunk 5]
```diff
@@ -83,4 +98,49 @@ export default class FormbuilderEditCodeController extends Controller {
...
+    };
+
+    if (!mapping[severity]) {
+      showErrorToasterMessage(
+        this.toaster,
+        this.intl.t('messages.error.unknownShaclSeverityType', {
+          severityType: severity,
+        })
+      );
```
Comment (JonasVanHoof) [inline app/controllers/formbuilder/edit/code.js:131]: We have mapped all the shacl [severity levels ](https://www.w3.org/TR/shacl/#severity)there are in the docs so this message will normally never show. BUT you never know

File: app/controllers/formbuilder/edit/code.js (hunk 2 above)
Comment (ZijneDwaasheid) [reply app/controllers/formbuilder/edit/code.js:60]: Yes, totally agree on this.

File: app/templates/formbuilder/edit/code.hbs [hunk 6]
```diff
@@ -1,43 +1,41 @@
...
+  <AuBodyContainer @scroll={{true}}>
     <AuContent
       @skin="small"
-      {{editor this.formCode (perform this.handleCodeChange)}}>
-    </AuContent>
+      {{editor this.formCode (perform this.handleCodeChange)}}
+    />
   </AuBodyContainer>
   {{#if this.features.USE_CONSOLE}}
-  <div class>
```
Comment (ZijneDwaasheid) [inline app/templates/formbuilder/edit/code.hbs:13]: the class can go?

File: translations/messages/nl-be.yaml [hunk 7]
```diff
@@ -31,6 +33,7 @@ messages:
     highestversionAvailable: "De laatste versie is v{version}"
     formIsValid: "Je formulier is correct"
     formIsInvalid: "Je formulier is niet correct. Vul al de velden in."
+    previewNotUpdatedBecauseOfErrorsInTtl: "Preview niet geupdate door errors in de ttl code"
```
Comment (ZijneDwaasheid) [inline translations/messages/nl-be.yaml:36]: geüpdatet :) als we correct willen zijn, of bijgewerkt

**Review by ZijneDwaasheid — APPROVED** [review]

File: app/templates/formbuilder/edit/code.hbs (hunk 6 above)
Comment (JonasVanHoof) [reply app/templates/formbuilder/edit/code.hbs:13]: Definitely :)

File: translations/messages/nl-be.yaml (hunk 7 above)
Comment (JonasVanHoof) [reply translations/messages/nl-be.yaml:36]: geüpdate ;p

File: translations/messages/nl-be.yaml [hunk 8]
```diff
@@ -31,6 +33,7 @@ messages:
     highestversionAvailable: "De laatste versie is v{version}"
     formIsValid: "Je formulier is correct"
     formIsInvalid: "Je formulier is niet correct. Vul al de velden in."
+    previewNotUpdatedBecauseOfErrorsInTtl: "Preview niet geüpdate door errors in de ttl code"
```
Comment (ZijneDwaasheid) [inline translations/messages/nl-be.yaml:36]: Almost, it's really "geüpdatet"
https://www.vlaanderen.be/team-taaladvies/taaladviezen/geupdatet-geupdatete-geupdate-geupdated

**Review by ZijneDwaasheid — APPROVED** [review]

**Pull request merged**